backend/lambdas/maintenance/db_cleanup/db_cleanup/tasks/scan.py

```diff
@@ -36,3 +39,18 @@ def _sbom_delete_check(scan: Scan) -> bool:
         ).count()
         > 0
     )
+
+
+def orphaned_s3_scan_data(log: Logger) -> None:
+    log.info("Cleaning up orphaned scan data from S3")
+    aws = AWSConnect()
+    files = aws.get_s3_file_list(prefix=SCANS_S3_KEY, s3_bucket=SCAN_DATA_S3_BUCKET, endpoint_url=SCAN_DATA_S3_ENDPOINT)
+    count = 0
+    for f in files:
+        scan_id = f.key.split("/")[1]  # Extract the scan ID from the S3 key: scans/<SCAN_ID>/...
```

Comment:
The only things that should be in this "directory" are valid scan IDs. If the ID is not valid (because it's not a UUID or because it's a UUID but there isn't a scan with that ID in the database) it is correct for it to be deleted below.